woocommerce/woocommerce-ios
# Support Order Fee Refunds

closes #5624 

# Why
Simple Payments as it is doesn't have any way to support refunds through the app, this PR looks to do the basic integration to allow those refunds.


# How

The main roadblock about refunding Simple Payments orders is related to the fact that every Simple Payment order uses Fees to insert the order total price, since the app doesn't support Fees refunds, this PR creates the exact same flow available for Shipping refunds, but for Fees instead. 

Every Order contains an array of `OrderFeeLine`, every Fee available inside this array is currently available for refund, when they're refunded Woo Core remove them from this array and they become available inside the Order as an `OrderItemRefund`.

# Screenshots

| Order containing fees (preferably from a Simple Payment order) | Issue Refund view with Fees Refund toggle disabled | Issue Refund view with Fees Refund toggle enabled | Order after fee refunds are complete |
| ------------- | ------------- | ------------- | ------------- |
| ![Simulator Screen Shot - iPhone 13 Pro Max - 2021-12-16 at 14 32 43](https://user-images.githubusercontent.com/5920403/146420764-535da85f-b96e-40d5-85f3-e384fe41c12c.png) | ![Simulator Screen Shot - iPhone 13 Pro Max - 2021-12-16 at 14 32 12](https://user-images.githubusercontent.com/5920403/146420788-22bd2667-ed45-4580-a7f9-e60ec3d19b2e.png) | ![Simulator Screen Shot - iPhone 13 Pro Max - 2021-12-16 at 14 33 02](https://user-images.githubusercontent.com/5920403/146420798-f9cb6c74-6aa5-40c8-a9d5-2edfcc54f709.png) | ![Simulator Screen Shot - iPhone 13 Pro Max - 2021-12-16 at 14 33 11](https://user-images.githubusercontent.com/5920403/146420809-7380ac8a-a55d-4930-a3cf-99f071ff0faf.png) |

# Testing Steps
1. Go to an Order containing Fees available for refunds
2. Click at `Issue Refunds` button
3. Verify that the `Refund Fees` Section is displayed as a single row containing a disabled toggle
4. Activate the Toggle and verify that the full Refund Fees section shows up, displaying currently the `subtotal`, `tax` and `total` fields, also verify that the `Next` button becomes clickable too.
5. Hit the `Next` button and verify if the Issue Confirmation view show up correctly as expected, hit the `Refund` button and check if the Refund happened as expected

---
